src/Tests.Microsoft.Security.Utilities.Core/WellKnownRegexPatternsTests.cs

```diff
@@ -26,6 +26,54 @@ public class WellKnownRegexPatternsTests
             "SEC101/109.AzureContainerRegistryLegacyKey"
         };
 
+        [TestMethod]
+        public void WellKnownRegexPatterns_AllRuleIdsAndNamesAreUnique()
+        {            
+            using var assertionScope = new AssertionScope();
+
+            var patterns = GetAllPatterns();
+
+            HashSet<string> ruleIds = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            HashSet<string> ruleNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
```suggestion
            HashSet<string> seenRuleIds = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
            HashSet<string> seenRuleNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

nit: these renames will make the loop code easier to read #Resolved